src/components/common/ExportToCsv/ExportToCsv.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { CSVLink } from 'react-csv'
+import { CSV_HEADERS, ExportToCsvProps, FILE_NAMES } from './constants'
+import { ReactComponent as ExportIcon } from '../../../assets/icons/ic-arrow-line-down.svg'
+import { ReactComponent as Success } from '../../../assets/icons/ic-success.svg'
+import { ReactComponent as Error } from '../../../assets/icons/ic-error-exclamation.svg'
+import { VisibleModal } from '../modals/VisibleModal'
+import './exportToCsv.scss'
+import { DetailsProgressing } from '../icons/Progressing'
+import moment from 'moment'
+import { Moment12HourExportFormat } from '../../../config'
+import Tippy from '@tippyjs/react'
+import { ConditionalWrap } from '../helpers/Helpers'
+
+export default function ExportToCsv({ apiPromise, fileName, className, disabled }: ExportToCsvProps) {
+    const [exportingData, setExportingData] = useState(false)
+    const [showExportingModal, setShowExportingModal] = useState(false)
+    const [errorExportingData, setErrorExportingData] = useState(false)
+    const [dataToExport, setDataToExport] = useState(null)
+    const [requestCancelled, setRequestCancelled] = useState(false)
+    const csvRef = useRef(null)
+    let abortController = null
+
+    useEffect(() => {
+        if (Array.isArray(dataToExport) && csvRef?.current && !requestCancelled) {
+            initiateDownload()
+        }
+    }, [dataToExport])
+
+    const generateDataToExport = async () => {
+        if (disabled) {
+            return
+        }
+
+        if (requestCancelled || errorExportingData) {
+            setRequestCancelled(false)
+            setErrorExportingData(false)
+        }
+
+        if (!exportingData) {
+            const fileNameKey = Object.keys(FILE_NAMES).find((key) => FILE_NAMES[key] === fileName)
+            console.info(`Started exporting data at ${moment().format('HH:mm:ss')}.`)
+            abortController = new AbortController()
+            setExportingData(true)
+            setShowExportingModal(true)
+
+            try {
+                const response = await apiPromise({ signal: abortController.signal })
+                setDataToExport(response)
+
+                console.info(
+                    `Completed data export of ${response.length} ${fileNameKey || ''} at ${moment().format(
+                        'HH:mm:ss',
+                    )}.`,
+                )
+            } catch (err) {
+                setErrorExportingData(true)
+
+                console.error(
```

Comment:
should we show this error to user? instead of just logging on cosole?

Reply:
As discussed, keeping this for now :)